### PR TITLE
Improved Docs

### PR DESCRIPTION
**Pull request by RandomDefaultUser**
I have mainly done three things:

- Fixed typos
- Rewrote API equations in LaTeX
- Overhauled the installation README.md; I think it is more accessible now to potential users. Feel free to correct me on this one though - I based it on my personal experience and how I would approach such a document, so I am open for suggestions.

### CODE REVIEW

**Comment by RandomDefaultUser** [comment]
This should be close #255.

**Review by DanielKotik — CHANGES_REQUESTED** [review]
That's great! I think this is a big step towards a cleaner, more accessible documentation.

File: docs/source/install/on_cluster.rst [hunk 1]
```diff
@@ -16,38 +16,30 @@ Installation: tips & tricks
...
-            - I got "Requirement already satisfied: oapackage in /home/fiedle09/.local/lib/python3.8/site-packages (from -r requirements.txt (line 9)) (2.6.8)" - I don't know how I got to this point, but the experience might differ for other people
-
-    3. Install Pytorch. I load CUDA for this.
+    2. Install Pytorch. I load CUDA for this.
 
         - ``module load cuda/10.2``
         - ``pip install torch``
 
-    4. Horovod needs CMAKE to work, so we need to load it along with the other dependencies required for it:
+    3. Then I install `mala` via
```
[the remark this answers is not in the document]
Comment (RandomDefaultUser) [reply docs/source/install/on_cluster.rst:32]: Good catch!

**Comment by DanielKotik** [comment]
Mhh, I am a bit puzzled. When I try to build locally, I get

```
❯ sphinx-build -W --keep-going -b html -d docs/_build/doctrees docs/source docs/_build/html
Sphinx v4.4.0 in Verwendung
making output directory... erledigt
myst v0.17.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=[], linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', disable_syntax=[], all_links_external=False, url_schemes=['http', 'https', 'mailto', 'ftp'], ref_domains=None, highlight_code_blocks=True, number_code_blocks=[], title_to_header=False, heading_anchors=None, heading_slug_func=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'], words_per_minute=200)
[autosummary] generating autosummary for: CONTRIBUTE.md, about.rst, about/background.rst, about/overview.md, api/mala.common.check_modules.rst, api/mala.common.json_serializable.rst, api/mala.common.parallelizer.rst, api/mala.common.parameters.rst, api/mala.common.rst, api/mala.datahandling.data_converter.rst, ..., install/on_cluster.rst, install/tested_systems.rst, installation.rst, usage.rst, usage/basicconcepts.rst, usage/hyperparameters.rst, usage/postprocessing.rst, usage/predictions.rst, usage/preprocessing.rst, usage/trainingmodel.rst
/home/kotik/GITHUB/work/mala_fork/mala/datahandling/data_repo.py:18: UserWarning: Environment variable MALA_DATA_REPO not set. You won't be able to run all examples and tests.
  warnings.warn(
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 64 source files that are out of date
updating environment: [new config] 64 added, 0 changed, 0 removed
reading sources... [100%] usage/trainingmodel                                                                                                                                                                                         
looking for now-outdated files... none found
pickling environment... erledigt
checking consistency... erledigt
preparing documents... erledigt
writing output... [100%] usage/trainingmodel                                                                                                                                                                                          
/home/kotik/GITHUB/work/mala_fork/docs/source/CONTRIBUTE.md:10: WARNING: 'myst' reference target not found: #branching-strategy
/home/kotik/GITHUB/work/mala_fork/docs/source/CONTRIBUTE.md:46: WARNING: 'myst' reference target not found: #developing-code
generating indices... genindex py-modindex erledigt
highlighting module code... [100%] mala.targets.target                                                                                                                                                                                
writing additional pages... search erledigt
copying images... [100%] ../../docs/source/img/logos/mala_horizontal.png                                                                                                                                                              
copying static files... erledigt
copying extra files... erledigt
dumping search index in English (code: en)... erledigt
dumping object inventory... erledigt
build mit Problemen beendet, 2 warnings.
```
Two wanings, that are turned into errors (due to the `-W`) flag and thus giving a non-zero exit code, that should lead to a failing pipeline.

**Comment by RandomDefaultUser** [comment]
> Mhh, I am a bit puzzled. When I try to build locally, I get
> 
> ```
> ❯ sphinx-build -W --keep-going -b html -d docs/_build/doctrees docs/source docs/_build/html
> Sphinx v4.4.0 in Verwendung
> making output directory... erledigt
> myst v0.17.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=[], linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', disable_syntax=[], all_links_external=False, url_schemes=['http', 'https', 'mailto', 'ftp'], ref_domains=None, highlight_code_blocks=True, number_code_blocks=[], title_to_header=False, heading_anchors=None, heading_slug_func=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'], words_per_minute=200)
> [autosummary] generating autosummary for: CONTRIBUTE.md, about.rst, about/background.rst, about/overview.md, api/mala.common.check_modules.rst, api/mala.common.json_serializable.rst, api/mala.common.parallelizer.rst, api/mala.common.parameters.rst, api/mala.common.rst, api/mala.datahandling.data_converter.rst, ..., install/on_cluster.rst, install/tested_systems.rst, installation.rst, usage.rst, usage/basicconcepts.rst, usage/hyperparameters.rst, usage/postprocessing.rst, usage/predictions.rst, usage/preprocessing.rst, usage/trainingmodel.rst
> /home/kotik/GITHUB/work/mala_fork/mala/datahandling/data_repo.py:18: UserWarning: Environment variable MALA_DATA_REPO not set. You won't be able to run all examples and tests.
>   warnings.warn(
> building [mo]: targets for 0 po files that are out of date
> building [html]: targets for 64 source files that are out of date
> updating environment: [new config] 64 added, 0 changed, 0 removed
> reading sources... [100%] usage/trainingmodel                                                                                                                                                                                         
> looking for now-outdated files... none found
> pickling environment... erledigt
> checking consistency... erledigt
> preparing documents... erledigt
> writing output... [100%] usage/trainingmodel                                                                                                                                                                                          
> /home/kotik/GITHUB/work/mala_fork/docs/source/CONTRIBUTE.md:10: WARNING: 'myst' reference target not found: #branching-strategy
> /home/kotik/GITHUB/work/mala_fork/docs/source/CONTRIBUTE.md:46: WARNING: 'myst' reference target not found: #developing-code
> generating indices... genindex py-modindex erledigt
> highlighting module code... [100%] mala.targets.target                                                                                                                                                                                
> writing additional pages... search erledigt
> copying images... [100%] ../../docs/source/img/logos/mala_horizontal.png                                                                                                                                                              
> copying static files... erledigt
> copying extra files... erledigt
> dumping search index in English (code: en)... erledigt
> dumping object inventory... erledigt
> build mit Problemen beendet, 2 warnings.
> ```
> 
> Two wanings, that are turned into errors (due to the `-W`) flag and thus giving a non-zero exit code, that should lead to a failing pipeline.

Could this be something with your local folders, maybe a rogue old file? I don't get these warnings on my machine.

**Comment by DanielKotik** [comment]
@RandomDefaultUser can you please resolve the conflicts?

**Comment by RandomDefaultUser** [comment]
Oh Sorry, I forgot about that. Will do.

**Comment by DanielKotik** [comment]
> > Mhh, I am a bit puzzled. When I try to build locally, I get
> > ```
> > ❯ sphinx-build -W --keep-going -b html -d docs/_build/doctrees docs/source docs/_build/html
> > Sphinx v4.4.0 in Verwendung
> > making output directory... erledigt
> > myst v0.17.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=[], linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', disable_syntax=[], all_links_external=False, url_schemes=['http', 'https', 'mailto', 'ftp'], ref_domains=None, highlight_code_blocks=True, number_code_blocks=[], title_to_header=False, heading_anchors=None, heading_slug_func=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'], words_per_minute=200)
> > [autosummary] generating autosummary for: CONTRIBUTE.md, about.rst, about/background.rst, about/overview.md, api/mala.common.check_modules.rst, api/mala.common.json_serializable.rst, api/mala.common.parallelizer.rst, api/mala.common.parameters.rst, api/mala.common.rst, api/mala.datahandling.data_converter.rst, ..., install/on_cluster.rst, install/tested_systems.rst, installation.rst, usage.rst, usage/basicconcepts.rst, usage/hyperparameters.rst, usage/postprocessing.rst, usage/predictions.rst, usage/preprocessing.rst, usage/trainingmodel.rst
> > /home/kotik/GITHUB/work/mala_fork/mala/datahandling/data_repo.py:18: UserWarning: Environment variable MALA_DATA_REPO not set. You won't be able to run all examples and tests.
> >   warnings.warn(
> > building [mo]: targets for 0 po files that are out of date
> > building [html]: targets for 64 source files that are out of date
> > updating environment: [new config] 64 added, 0 changed, 0 removed
> > reading sources... [100%] usage/trainingmodel                                                                                                                                                                                         
> > looking for now-outdated files... none found
> > pickling environment... erledigt
> > checking consistency... erledigt
> > preparing documents... erledigt
> > writing output... [100%] usage/trainingmodel                                                                                                                                                                                          
> > /home/kotik/GITHUB/work/mala_fork/docs/source/CONTRIBUTE.md:10: WARNING: 'myst' reference target not found: #branching-strategy
> > /home/kotik/GITHUB/work/mala_fork/docs/source/CONTRIBUTE.md:46: WARNING: 'myst' reference target not found: #developing-code
> > generating indices... genindex py-modindex erledigt
> > highlighting module code... [100%] mala.targets.target                                                                                                                                                                                
> > writing additional pages... search erledigt
> > copying images... [100%] ../../docs/source/img/logos/mala_horizontal.png                                                                                                                                                              
> > copying static files... erledigt
> > copying extra files... erledigt
> > dumping search index in English (code: en)... erledigt
> > dumping object inventory... erledigt
> > build mit Problemen beendet, 2 warnings.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Two wanings, that are turned into errors (due to the `-W`) flag and thus giving a non-zero exit code, that should lead to a failing pipeline.
> 
> Could this be something with your local folders, maybe a rogue old file? I don't get these warnings on my machine.

I found the reason. I was just using my usual environment in order to build the docs, which uses `myst-parser-0.17.0`. Using a `venv` with exactly what we have specified in `docs/requirements.txt` (i.e. `myst-parser-0.140`) I don't see this issue.

This brings 
- #279

to my attention again. We should update the docs requirements to recent versions of Sphinx etc and fix possible issues in a separate PR.

**Comment by DanielKotik** [comment]
Here are some formatting issues:
grafik
All those line should look equal.

**Comment by RandomDefaultUser** [comment]
> Here are some formatting issues: grafik All those line should look similar. There is a linebreak issue.

I fixed the linebreak issue.

File: requirements.txt [hunk 2]
```diff
@@ -6,5 +6,4 @@ mpmath
 numpy
 optuna
 scipy
-pandas
```
Comment (DanielKotik) [inline requirements.txt:9]: In what way is removing `pandas` related to improving the the docs? I suggest to leave pandas as a dependency in this PR and, if obsolete, remove it in a additional PR and then also in `mala_cpu_base_environment.yml` and `mala_cpu_environment.yml` with an explanation why `pandas` is not needed any more.

**Review by DanielKotik — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (RandomDefaultUser) [reply requirements.txt:9]: I rewrote the installation instructions, and removed `pandas` there. Afterwards I realized I should keep it consistent by also removing it from the `requirements.txt`. It is definitely obsolete. I can revert the `requirements.txt` and make a new PR.

**Pull request merged**